pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -220,6 +220,11 @@ func (a *ApplicationBackupController) createBackupLocationPath(backup *stork_api
 
 // handle updates for ApplicationBackup objects
 func (a *ApplicationBackupController) handle(ctx context.Context, backup *stork_api.ApplicationBackup) error {
+	// If the backup is already in final stage, return with out doing anything.
+	if backup.Status.Stage == stork_api.ApplicationBackupStageFinal {
```

Comment:
lets move it after the deletion timestamp check

Reply:
Fixed